ocaml/xenopsd/xc/xenctrlext.ml

```diff
@@ -82,3 +82,65 @@ type cputopo = {core: int; socket: int; node: int}
 external numainfo : handle -> numainfo = "stub_xenctrlext_numainfo"
 
 external cputopoinfo : handle -> cputopo array = "stub_xenctrlext_cputopoinfo"
+
+module Xenforeignmemory : sig
+  type handle
+
+  type mapping =
```

Comment:
maybe it'd be best to hide the type and provide some functions to read/write the data from a string.
Then we could check that it hasn't been freed yet, and the unsafe interface would only be internal to this module: it'd be impossible to use it incorrectly from the outside: we'd raise an exception.

Reply:
I agree, I think this type should be abstract.

Reply:
The lifetime of the mapping is already limited by the scope of the function `with_mapping`. 

While this interface is unsafe, it's not clear to me how it's going to be used. I would rather change the interface later on once we write code in xenops that handles these arrays to better know the needs for a safe interface